pkg/console/controllers/clidownloads/controller.go

```diff
@@ -0,0 +1,260 @@
+	}
+
+	operatorConfig, err := c.operatorConfigClient.Get(api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	updatedOperatorConfig := operatorConfig.DeepCopy()
+
+	consoleRoute, err := c.routeClient.Routes(api.TargetNamespace).Get(api.OpenShiftConsoleRouteName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	host := routesub.GetCanonicalHost(consoleRoute)
+	ocConsoleCLIDownloads := PlatformBasedOCConsoleCLIDownloads(host, "amd64", ocCLIDownloadsCustomResourceName)
+	_, ocCLIDownloadsErrReason, ocCLIDownloadsErr := CreateOrUpdate(c.consoleCliDownloadsClient, ocConsoleCLIDownloads)
+	status.HandleDegraded(updatedOperatorConfig, "ConsoleCliDownloadsSync", ocCLIDownloadsErrReason, ocCLIDownloadsErr)
+	if ocCLIDownloadsErr != nil {
+		return ocCLIDownloadsErr
+	}
+
+	_, odoCLIDownloadsErrReason, odoCLIDownloadsErr := CreateOrUpdate(c.consoleCliDownloadsClient, ODOConsoleCLIDownloads())
+	status.HandleDegraded(updatedOperatorConfig, "ConsoleCliDownloadsSync", odoCLIDownloadsErrReason, odoCLIDownloadsErr)
+	if odoCLIDownloadsErr != nil {
+		return odoCLIDownloadsErr
+	}
+
+	return nil
+}
+
+func PlatformBasedOCConsoleCLIDownloads(host, arch, cliDownloadsName string) *v1.ConsoleCLIDownload {
+	baseURL := fmt.Sprintf("%s/%s", util.HTTPS(host), arch)
+	getPlatformURL := func(platform, archiveType string) string {
+		return fmt.Sprintf("%s/%s/%s", baseURL, platform, archiveType)
+	}
+
+	return &v1.ConsoleCLIDownload{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cliDownloadsName,
+		},
+		Spec: v1.ConsoleCLIDownloadSpec{
+			Description: `With the OpenShift command line interface, you can create applications and manage OpenShift projects from a terminal.
+
+The oc binary offers the same capabilities as the kubectl binary, but it is further extended to natively support OpenShift Container Platform features.
+`,
+			DisplayName: "oc - OpenShift Command Line Interface (CLI)",
+			Links: []v1.Link{
+				{
+					Href: getPlatformURL(LinuxPlatformKey, "oc.tar"),
+					Text: "Download oc for Linux",
+				},
+				{
+					Href: getPlatformURL(MacPlatformKey, "oc.tar"),
+					Text: "Download oc for Mac",
+				},
+				{
+					Href: getPlatformURL(WindowsPlatformKey, "oc.tar"),
+					Text: "Download oc for Windows",
+				},
+			},
+		},
+	}
+}
+
+func ODOConsoleCLIDownloads() *v1.ConsoleCLIDownload {
+	return &v1.ConsoleCLIDownload{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: odoCLIDownloadsCustomResourceName,
+		},
+		Spec: v1.ConsoleCLIDownloadSpec{
+			Description: `OpenShift Do (odo) is a fast, iterative, and straightforward CLI tool for developers who write, build, and deploy applications on OpenShift.
+
+odo abstracts away complex Kubernetes and OpenShift concepts, thus allowing developers to focus on what is most important to them: code.
+`,
+			DisplayName: "odo - Developer-focused CLI for OpenShift",
+			Links: []v1.Link{
+				{
+					Href: "https://mirror.openshift.com/pub/openshift-v4/clients/odo/latest/",
```

Comment:
Are we good with this `href` being hard-coded?  I imagine it doesn't really change?  Any concerns about versions?

Reply:
good question, @spadgett do you know if there is any chance that this shall change ?
If so can add and entry to the operator config and read it from there ?

Reply:
I'm not sure about odo API compatibility, but at some point, we might want to lock to a specific odo version for a specific openshift version. It is hard-coded to latest today.

@mohammedzee1000 @kadel We now have an extension point in console for adding to the command-line download page. Is there another operator that should own creating the CR for the odo download? If console owns it, what version should we link to?

Reply:
Currently, the odo is aiming to be compatible with 3.9 and up. odo is trying to be compatible with every supported OCP version. Using the latest should be fine.

There is no odo operator if the console can own this that would be great.

Reply:
@spadgett I guess than our operator will managed `odo` as well

Reply:
@kadel Should we point to odo `latest`? Or a version like `v0.0.20`?

Reply:
I'm fine managing it if we can point to `latest`.  If we get to a point where we need to keep up with version numbers, would we want to revisit the mechanism for that?

Reply:
Also, just validating, `odo` is not built for several `platforms`?  `linux` only, I assume?  Should this be clarifiedin the text?

Reply:
> Also, just validating, odo is not built for several platforms? linux only, I assume? Should this be clarifiedin the text?

we probably should

Reply:
> Also, just validating, `odo` is not built for several `platforms`? `linux` only, I assume? Should this be clarifiedin the text?

odo is built for all 3 platforms (windows, mac, linux) all 3 of them are available at https://mirror.openshift.com/pub/openshift-v4/clients/odo/latest/

Reply:
> I'm fine managing it if we can point to `latest`. If we get to a point where we need to keep up with version numbers, would we want to revisit the mechanism for that?

Right now the only time where I see that we would want to keep up with the versions is when there are some big changes in core openshfit or k8s apis. Wouldn't that also result in OpenShift v5?

Reply:
Should this then mirror the links we generate from console, by listing the 3 links separately?

```
odo-linux-amd64.tar.gz  
odo-darwin-amd64.tar.gz  
odo-windows-amd64.exe
```

Reply:
![Screen Shot 2019-10-07 at 12 37 38 PM](https://user-images.githubusercontent.com/280512/66623814-68bae000-ebbb-11e9-859f-2e82a83dc2d0.png)

Reply:
This would imply that the file names are stable, but is likely a better experience if we can confirm.

Reply:
I think that it might be safer to go with a link to the latest directory for now.
We are just preparing the 1.0 release, and it might change a little bit. After the release it will be stable, but I can't guarantee the exact filenames right now.

Reply:
ok that works.

Reply:
I would probably recommend the individual files as a better experience, once you are set.